core/Rename.cpp

```diff
@@ -271,18 +285,28 @@ namespace olympia
                 const auto & dests = renaming_inst->getDestOpInfoList();
                 for(const auto & src : srcs)
                 {
+                    const auto rf  = olympia::coreutils::determineRegisterFile(src);
+                    const auto num = src.field_value;
+                    // we check if src is RF_INTEGER x0, if so, we skip rename
+                    const bool is_x0 = (num == 0 && rf == core_types::RF_INTEGER);
+                    if (is_x0) {
+                        // if x0 is a data operand for LSU op, we need to set it in DataReg when we check in LSU
+                        // so we can still check the scoreboard, which will always return back ready for x0
```

Comment:
I don't think this is necessary.  The SB should never be checked for x0 -- in other words, you don't need to set that bit as a source for this instruction in the bit mask.

Reply:
Resolved in [650f6c6](https://github.com/riscv-software-src/riscv-perf-model/pull/120/commits/650f6c6b24edbaac804d3a6219ddc783a829b8cc), added logic to just check `is_x0` and avoid even checking scoreboard bits.